src/p2p/net_node.inl

```diff
@@ -1116,20 +1185,36 @@ namespace nodetool
   }
   //-----------------------------------------------------------------------------------
   template<class t_payload_net_handler>
-  bool node_server<t_payload_net_handler>::make_expected_connections_count(bool white_list, size_t expected_connections)
+  bool node_server<t_payload_net_handler>::make_expected_connections_count(PeerType peer_type, size_t expected_connections)
   {
     if (m_offline)
       return true;
 
+    std::vector<anchor_peerlist_entry> apl;
+
+    if (peer_type == anchor) {
+      m_peerlist.get_and_empty_anchor_peerlist(apl);
+    }
```

Comment:
Ah, I get it. It's emptied, and then refilled with the ones still live we know of. OK, thanks.